src/fill-zulipdb-with-msgs-deploy.mjs

```diff
@@ -0,0 +1,191 @@
+          for (db <- dbCh) {
+            db!("select", "${table}", [${ids}], *selectCh) |
+            for (@keys <- selectCh) {
+              return!(("select", keys))
+            }
+          }
+        }
+      }
+    `;
+  }
+
+  function getTablesNames(result) {
+    const par = result.postblockdataList[0];
+    const rhoResult = rhoParToJson(par);
+    return rhoResult[1];
+  }
+  
+  function getKeys(result) {
+    const par = result.postblockdataList[0];
+    const rhoResult = rhoParToJson(par);
+    return rhoResult[1][1];
+  }
+
+  function getDatafromAST(dataList){
+    let messages = [];
+    for (let msgData of dataList){
+      let tupleBody = msgData.exprsList[0]?.eTupleBody;  
+      if (tupleBody != undefined){
+        for (let list of tupleBody.psList){
+          if (list.exprsList[0]?.eMapBody != undefined){
+            let dataList = list.exprsList[0]?.eMapBody.kvsList;
+            for (let data of dataList){
+              let mapBody = data.value.exprsList[0]?.eMapBody;        
+              if (mapBody != undefined){
+                let msg = {};
+                msg["id"] = data.key.exprsList[0]?.gInt;
+            
+                for (let dictEl of mapBody.kvsList){    
+                  if (dictEl.value){
+                    let key = dictEl.key.exprsList[0]?.gString;
+                    let valueSt = dictEl.value.exprsList[0]?.gString;
+                    let valueInt = dictEl.value.exprsList[0]?.gInt;
+                    let valueBool = dictEl.value.exprsList[0]?.gBool;
+                    let value =  valueSt || valueInt || valueBool;
+                    msg[key] = value;
+                  }
+                }
+                messages.push(msg);
+              }
+            }
+          }
+        }
+      }
+    }
+    return messages;
+  }
+
+  async function saveDataToDB(sql, tableName, data, keys, returnIds) {
+    try {
+      const insertedRecords = await sql`INSERT INTO ${sql(tableName)} ${sql(data, ...keys)} returning id`;
+      console.log("# of records inserted:", data.length);
+      if (returnIds){
+        return insertedRecords;
+      }
+    } catch (error) {
+      console.error("Postgres insert error: "+error);
+      await sql.end();
+    }
+  }
+
+  // RNode connection
+  const { exploratoryDeploy } = rnodeService(env, grpcLib);
+  let insertedIds = [];
+  // Get tables to get data
+  let {result} = await exploratoryDeploy({term: getTables()});
+  const tablesNames = getTablesNames(result);
+  
+  for (let table of tablesNames){
+    let returnIds = false;
+    // Get ids for tables
+    let {result} = await exploratoryDeploy({term: getTableKeys(table)});
+    const recordKeys = getKeys(result);
+
+    // Get data by ids
+    let {result: resultRecords} = await exploratoryDeploy({term: getRecordInDB(table, recordKeys)});
+    const tableDataList = resultRecords.postblockdataList[0]?.exprsList[0]?.eTupleBody.psList;
+    let tableData = getDatafromAST(tableDataList);
```

Comment:
Can this also be replaced with `rhoParToJson`?

Reply:
I'm getting 
```@tgrospic/rnode-grpc-js/dist/rho-json.js:27
if (!!((_a = input.exprsList) === null || _a === void 0 ? void 0 : _a.length)) {
                       ^

TypeError: Cannot read property 'exprsList' of undefined
```

if I run it sorrounded by `try catch ` I get the same error but it also shows 
```
RESULT [
  'select',
  [
    true,
    { '7': { flags: 1, id: 7, message_id: 6, user_profile_id: 8 } }
  ]
]
```

So I guess `rhoParToJson` needs to be improved in order to parse data with a "deeper" structure such as the one from messages data (check `getDatafromAST()`).

Reply:
You are right, I found the problem in `rhoParToJson` when null/undefined is the value in a Map. We can resolve this in later PR.